invenio_app_ils/search_permissions.py

```diff
@@ -63,11 +63,18 @@ def ils_search_factory(self, search, validator=None):
     :returns: Tuple with search instance and URL arguments.
     """
 
-    def query_parser(qstr=None):
+    def query_parser(search, qstr=None):
```

Comment:
You can eventually use the `search` param from outside... not really important
```suggestion
    def query_parser(qstr=None):
```

Reply:
I changed it because it passes through a validator before query_parser is even called and basically validator can return changed search instance